File: Model/AssetResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Auctus.Model
{
    public class AssetResponse
    {
        public int AssetId { get; set; } 
        public string Name { get; set; }
        public string Code { get; set; }
        public int TotalRatings { get; set; }
        public int TotalAdvisors { get; set; }
        public int Mode { get; set; }
        public double LastValue { get; set; }
        public bool ShortSellingEnabled { get; set; }
        public double? Variation24h { get; set; }
        public double? Variation7d { get; set; }
        public double? Variation30d { get; set; }
        public double? MarketCap { get; set; }
        public double? CirculatingSupply { get; set; }
        public int? NumberOfFollowers { get; set; }
        public bool? Following { get; set; }
        public PairResponse Pair { get; set; }
        public List<RecommendationDistributionResponse> RecommendationDistribution { get; set; } = new List<RecommendationDistributionResponse>();
        public List<RecommendationDistributionResponse> ReportRecommendationDistribution { get; set; } = new List<RecommendationDistributionResponse>();
        
        public class ValuesResponse
        {
            public DateTime Date { get; set; }
            public double Value { get; set; }
        }
        public class PairResponse
        {
            public string Symbol { get; set; }
            public string MultipliedSymbol { get; set; }
            public string Preffix { get; set; }
            public string Suffix { get; set; }
        }
    }
}

File: Web/src/app/model/asset/assetResponse.ts
import { RecommendationDistributionResponse } from "../recommendationDistributionResponse";
import { AdvisorResponse } from "../advisor/advisorResponse";
import { ReportResponse } from "./reportResponse";
import { EventResponse } from "./eventResponse";

export class AssetResponse {
  assetId: number;
  name: string;
  code: string;
  totalRatings?: number;
  totalAdvisors?: number;
  lastValue?: number;
  variation24h?: number;
  variation7d?: number;
  variation30d?: number;
  numberOfFollowers?: number;
  following?:boolean;
  mode: number;
  pair: PairResponse;
  recommendationDistribution: RecommendationDistributionResponse[];
  reportRecommendationDistribution: RecommendationDistributionResponse[];
  assetAdvisor?: AssetAdvisorResponse[];
  advices?: AdviceResponse[];
  advisors?: AdvisorResponse[];
  reports?: ReportResponse[];
  events?: EventResponse[];
}

export class AssetAdvisorResponse
{
  userId: number;
  successRate: number;
  averageReturn: number;
  currentReturn: number;
  totalRatings: number;
  lastAdviceDate: Date;
  lastAdviceType: number;
  lastAdviceMode: number;
  lastAdviceOperationType: number;
  lastAdviceTargetPrice?: number;
  lastAdviceStopLoss?: number;
  advices: AdviceResponse[];
}

export class ValuesResponse{
  date: string;
  value: number;
}

export class AdviceResponse{
  date: string;
  adviceType: number;
  assetValue : number;
  operationType: number;
  targetPrice?: number;
  stopLoss?: number;
}

export class PairResponse{
  symbol: string;
  multipliedSymbol: string;
  preffix: string;
  suffix: string;
}